src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplateToken.cs

```diff
@@ -1,32 +1,273 @@
+                }
+            }
+
+            if (builtin == null)
+            {
+                return new ExpressionToken(parts);
+            }
+            else
+            {
+                return new ExpressionTokenOrBuiltin(parts, builtin);
+            }
+        }
+
+        public abstract string Evaluate(IReadOnlyDictionary<string, object> bindingData);
+
+        protected abstract void BuildCapturePattern(StringBuilder builder);
+
+        // Create a regex that will populate values of parameters by matching the template against an input string. 
+        // this is used by triggers to populate binding data. 
+        internal static string BuildCapturePattern(IEnumerable<BindingTemplateToken> tokens)
+        {
+            StringBuilder builder = new StringBuilder("^");
+
+            foreach (var token in tokens)
+            {
+                token.BuildCapturePattern(builder);
+            }
+
+            return builder.Append("$").ToString();
+        }
 
-        public BindingTemplateToken(string value, bool isParameter)
+        // Represents a literal segment.
+        private class LiteralToken : BindingTemplateToken
         {
-            _value = value;
-            _isParameter = isParameter;
+            private readonly string _value;
+            public LiteralToken(string value)
+            {
+                _value = value;
+            }
+
+            public override string AsLiteral
+            {
+                get
+                {
+                    return _value;
+                }
+            }
+
+            public override string ParameterName
+            {
+                get
+                {
+                    return null;
+                }
+            }
+            
+            public override string Evaluate(IReadOnlyDictionary<string, object> bindingData)
+            {
+                return _value;
+            }
+
+            protected override void BuildCapturePattern(StringBuilder builder)
+            {
+                builder.Append(Regex.Escape(this._value));
+            }
         }
 
-        public string Value 
-        { 
-            get { return _value; } 
+        // Represent an expression. 
+        //    "literal{expr}literal" 
+        private class ExpressionToken : BindingTemplateToken
+        {
+            private readonly string[] _parts;
+
+            public ExpressionToken(string[] parts)
+            {
+                _parts = parts;
+            }
+
+            public override string AsLiteral
+            {
+                get
+                {
+                    return null;
+                }
+            }
+
+            public override string ParameterName
+            {
+                get
+                {
+                    return this._parts[0];
+                }
+            }
+
+            protected override void BuildCapturePattern(StringBuilder builder)
```

Comment:
No. The caller up the stack has a bigger comment. This is used on triggers. IE, 
    [BlobTrigger("container/{name}") 
Will match against he incoming blob path and assign a value to 'name'


---
In reply to: [112484166](https://github.com/Azure/azure-webjobs-sdk/pull/1117#discussion_r112484166) [](ancestors = 112484166)